isis/src/docsys/documents/DeveloperAddFeature/DeveloperAddFeature.xml

```diff
@@ -0,0 +1,187 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<documentation xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://isis.astrogeology.usgs.gov/Schemas/Documentation/documentation.xsd">
+  <files>
+    <file>
+      <body>
+        <h2><a name="Introduction">Introduction</a></h2>
+          <p>
+            This is a step-by-step guide describing the process for adding a new feature to ISIS.
+            Since ISIS is made up of many application programs, when we say new feature we are referring adding
+            a new program to ISIS. Each
+            program in ISIS is focussed on a narrow subject, for example, isis2std is focussed on exporting ISIS cubes
+            to standard image formats such as jpeg, tif, or png.
+          </p>
+          <p>
+            ISIS applications are written in C++ and are supported on Ubuntu Linux and Mac OS.
+          </p>
+
+        <h2><a name="Resources">Resources</a></h2>
+          <p>
+            The <a href="https://isis.astrogeology.usgs.gov/Object/Developer/index.html>ISIS library"</a> contains 
+            hundreds of support classes and functions. Every application takes advantage of
+            these support routines to accomplish its tasks. We encourage you to get familiar with the library before 
+            starting to write your first application.
+          </p>
+          <p>
+            There are over 400 <a href="https://isis.astrogeology.usgs.gov/Application/index.html"ISIS applications</a>.
+            We suggest you get familiar with as many of these as possible, so you don't duplicate a feature that has already 
+            been implemented, and to use an existing application as a template for your new feature.
+          </p>
+          <p>
+            ISIS applications are linked against many external libraries. We encourage you to become familiar with these 
+            external libraries. One place to start is the environment.yml file located at the root of the ISIS source
```

Comment:
I would say that "a list of the external libraries that ISIS uses directly can be found in the environment.yml file"